exporter/exporterhelper/logs.go

```diff
@@ -85,18 +85,24 @@ func NewLogsExporter(
 
 	bs := fromOptions(options...)
 	be := newBaseExporter(cfg, logger, bs)
+	obsrep := obsreport.NewExporter(obsreport.ExporterSettings{
+		Level:      configtelemetry.GetMetricsLevelFlagValue(),
+		ExporterID: cfg.ID(),
+	})
 	be.wrapConsumerSender(func(nextSender requestSender) requestSender {
 		return &logsExporterWithObservability{
-			obsrep: obsreport.NewExporter(obsreport.ExporterSettings{
-				Level:      configtelemetry.GetMetricsLevelFlagValue(),
-				ExporterID: cfg.ID(),
-			}),
+			obsrep:     obsrep,
 			nextSender: nextSender,
 		}
 	})
 
 	lc, err := consumerhelper.NewLogs(func(ctx context.Context, ld pdata.Logs) error {
-		return be.sender.send(newLogsRequest(ctx, ld, pusher))
+		req := newLogsRequest(ctx, ld, pusher)
+		err := be.sender.send(req)
+		if err == errSendingQueueIsFull {
```

Comment:
Should we use https://golang.org/pkg/errors/#Is?

Reply:
Thanks. applied